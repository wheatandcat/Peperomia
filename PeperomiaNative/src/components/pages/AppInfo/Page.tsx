import React, { Component } from "react";
import { StyleSheet, View, Text, Image } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import AppIntroSlider from "react-native-app-intro-slider";
import { whenIPhoneSE } from "../../../lib/responsive";
import theme from "../../../config/theme";

interface Props {
  onDone: () => void;
}

const styles = StyleSheet.create({
  buttonCircle: {
    width: 40,
    height: 40,
    backgroundColor: "rgba(0, 0, 0, .2)",
    borderRadius: 20,
    justifyContent: "center",
    alignItems: "center"
  }
});

type Slide = {
  key: string;
  title: string;
  text: string;
  image: any;
  imageWidth: number;
  titleColor: string;
  textColor: string;
  backgroundColor: string;
};

const slides: Slide[] = [
  {
    key: "step1",
    title: "予定を管理",
    text: "ペペロミアは予定管理アプリです\n簡単な操作で予定を作成",
    image: require("../../../img/intro_home.png"),
    imageWidth: 250,
    titleColor: theme().color.main,
    textColor: theme().color.main,
    backgroundColor: theme().color.lightGreen
  },
  {
    key: "step2",
    title: "予定を整理",
    text:
      "タイトルをつけると自動でアイコンを設定\n見やすい予定表を作成しよう！",
    image: require("../../../img/intro_plan2.png"),
    imageWidth: 250,
    titleColor: theme().color.lightGreen,
    textColor: theme().color.lightGreen,
    backgroundColor: theme().color.main
  },
  {
    key: "step3",
    title: "予定を共有",
    text: "作成した予定は\nブラウザから誰にでも共有可能",
    image: require("../../../img/intro_share.png"),
    imageWidth: 300,
    titleColor: theme().color.main,
    textColor: theme().color.main,
    backgroundColor: theme().color.lightGreen
  },
  {
    key: "step4",
    title: "ようこそ！！",
    text: "ペペロミアを使って予定を作っていこう！",
    image: require("../../../img/icon.png"),
    imageWidth: 200,
    titleColor: theme().color.white,
    textColor: theme().color.highLightGray,
    backgroundColor: theme().color.main
  }
];

export default class extends Component<Props> {
  _renderNextButton = () => {
    return (
      <View style={styles.buttonCircle}>
        <Ionicons
          name="md-arrow-round-forward"
          color="rgba(255, 255, 255, .9)"
          size={24}
          style={{
            backgroundColor: "transparent"
          }}
        />
      </View>
    );
  };

  _renderDoneButton = () => {
    return (
      <View style={styles.buttonCircle}>
        <Ionicons
          name="md-checkmark"
          color="rgba(255, 255, 255, .9)"
          size={24}
          style={{
            backgroundColor: "transparent"
          }}
        />
      </View>
    );
  };

  _renderItem = ({ item }: { item: Slide }) => (
    <View
      style={{
        backgroundColor: item.backgroundColor,
        height: "100%",
        paddingTop: whenIPhoneSE(30, 150)
      }}
    >
      <View
        style={{
          paddingTop: 70,
          height: 300,
          paddingLeft: 10,
          alignItems: "center",
          justifyContent: "center"
        }}
      >
        <Image
          source={item.image}
          style={{ width: item.imageWidth }}
          resizeMode="contain"
        />
      </View>
      <View
        style={{
          justifyContent: "center",
          paddingLeft: 50,
          paddingTop: 50
        }}
      >
        <Text
          style={{
            fontSize: 25,
            fontWeight: "600",
            color: item.titleColor
          }}
        >
          {item.title}
        </Text>
      </View>
      <View
        style={{
          justifyContent: "center",
          alignItems: "flex-start",
          paddingLeft: 50,
          paddingTop: 20
        }}
      >
        {item.text.split("\n").map((val: string) => (
          <Text
            style={{
              fontSize: 16,
              color: item.textColor,
              paddingBottom: 5
            }}
            key={val}
          >
            {val}
          </Text>
        ))}
      </View>
    </View>
  );

  _onDone = () => {
    this.props.onDone();
  };

  render() {
    return (
      <AppIntroSlider
        slides={slides}
        onDone={this._onDone}
        renderItem={this._renderItem}
        renderDoneButton={this._renderDoneButton}
        renderNextButton={this._renderNextButton}
      />
    );
  }
}
