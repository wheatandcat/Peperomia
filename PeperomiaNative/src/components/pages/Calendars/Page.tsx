import React, { Component } from "react";
import {
  Text,
  View,
  Dimensions,
  StyleSheet,
  Image,
  TouchableOpacity,
  ActivityIndicator,
  Animated,
  SafeAreaView,
  ScrollView
} from "react-native";
import { Calendar, LocaleConfig } from "react-native-calendars";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import GestureRecognizer from "react-native-swipe-gestures";
import dayjs from "dayjs";
import advancedFormat from "dayjs/plugin/advancedFormat";
import "dayjs/locale/ja";
import Color from "color";
import { SelectCalendar } from "../../../lib/db/calendar";
import theme from "../../../config/theme";
import GlobalStyles from "../../../GlobalStyles";
import ImageDay from "../../organisms/Calendars/Image";

dayjs.extend(advancedFormat);

const width = Dimensions.get("window").width;
const AnimatedSafeAreaView = Animated.createAnimatedComponent(SafeAreaView);

LocaleConfig.locales["jp"] = {
  monthNames: [
    "1月",
    "2月",
    "3月",
    "4月",
    "5月",
    "6月",
    "7月",
    "8月",
    "9月",
    "10月",
    "11月",
    "12月"
  ],
  monthNamesShort: [
    "1月",
    "2月",
    "3月",
    "4月",
    "5月",
    "6月",
    "7月",
    "8月",
    "9月",
    "10月",
    "11月",
    "12月"
  ],
  dayNames: ["日", "月", "火", "水", "木", "金", "土"],
  dayNamesShort: ["日", "月", "火", "水", "木", "金", "土"]
};
LocaleConfig.defaultLocale = "jp";

const images = [
  require("../../../img/months/january.png"),
  require("../../../img/months/february.png"),
  require("../../../img/months/march.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/october.png"),
  require("../../../img/months/november.png"),
  require("../../../img/months/december.png")
];

type Props = {
  loading: boolean;
  calendars: SelectCalendar[];
  onCreate: (date: string) => void;
  onSchedule: (id: number, title: string) => void;
};

type State = {
  currentDate: string;
  count: number;
  backgroundColor: any;
};

const backgroundColors = [
  "#FDDACB",
  "#FBEDF2",
  "#DCF6FA",
  "#fdeff2",
  "#badcad",
  theme().color.dodgerBlue,
  "#a0d8ef",
  "#f2f2b0",
  theme().color.beige,
  "#E3E3FE",
  "#FFE7D0",
  "#BFD3B7"
];

const config = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80
};

export default class extends Component<Props, State> {
  state = {
    currentDate: dayjs().format("YYYY-MM-01"),
    count: 0,
    backgroundColor: new Animated.Value(dayjs().month())
  };

  calendar: any;
  componentDidMount() {
    const currentMonth = dayjs(this.state.currentDate).month();

    Animated.timing(this.state.backgroundColor, {
      toValue: currentMonth
    }).start();
  }

  onNextMonth = () => {
    if (this.state.count === 3) {
      return;
    }

    const count = this.state.count + 1;
    const currentMonth = dayjs(this.state.currentDate)
      .add(count, "month")
      .month();

    this.setState({
      count
    });

    if (currentMonth === 0) {
      Animated.timing(this.state.backgroundColor, {
        toValue: 12
      }).start(props => {
        if (props.finished) {
          this.setState({
            backgroundColor: new Animated.Value(0)
          });
        }
      });
    } else {
      Animated.timing(this.state.backgroundColor, {
        toValue: currentMonth
      }).start();
    }
    this.calendar.addMonth(1);
  };

  onPrevMonth = () => {
    if (this.state.count === -3) {
      return;
    }

    const count = this.state.count - 1;
    const currentMonth = dayjs(this.state.currentDate)
      .add(count, "month")
      .month();

    this.setState({
      count
    });

    if (currentMonth === 11) {
      Animated.timing(this.state.backgroundColor, {
        toValue: -1
      }).start(props => {
        if (props.finished) {
          this.setState({
            backgroundColor: new Animated.Value(12)
          });
        }
      });
    } else {
      Animated.timing(this.state.backgroundColor, {
        toValue: currentMonth
      }).start();
    }
    this.calendar.addMonth(-1);
  };

  render() {
    const backgroundColor = this.state.backgroundColor.interpolate({
      inputRange: [-1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
      outputRange: [
        backgroundColors[11],
        ...backgroundColors,
        backgroundColors[0]
      ]
    });

    const currentMonth = dayjs(this.state.currentDate)
      .add(this.state.count, "month")
      .month();

    const animationStyle = {
      backgroundColor
    };

    return (
      <ScrollView
        style={{
          height: "100%",
          width: "100%",
          backgroundColor: backgroundColors[currentMonth]
        }}
      >
        <AnimatedSafeAreaView
          style={[GlobalStyles.droidSafeArea, { flex: 0 }, animationStyle]}
        />
        <GestureRecognizer
          onSwipeLeft={this.onNextMonth}
          onSwipeRight={this.onPrevMonth}
          config={config}
        >
          <AnimatedSafeAreaView style={[styles.root, animationStyle]}>
            <View style={styles.headerContainer}>
              {this.state.count === -2 ? (
                <View />
              ) : (
                <TouchableOpacity onPress={this.onPrevMonth}>
                  <MaterialCommunityIcons
                    name="chevron-left"
                    size={30}
                    color={theme().color.main}
                  />
                </TouchableOpacity>
              )}

              <View style={styles.yearContainer}>
                <Text style={styles.year}>
                  {dayjs(this.state.currentDate)
                    .add(this.state.count, "month")
                    .format("YYYY年MM月")}
                </Text>
              </View>

              {this.state.count === 3 ? (
                <View />
              ) : (
                <TouchableOpacity onPress={this.onNextMonth}>
                  <MaterialCommunityIcons
                    name="chevron-right"
                    size={30}
                    color={theme().color.main}
                  />
                </TouchableOpacity>
              )}
            </View>
            <View style={styles.imageContainer}>
              <Image source={images[currentMonth]} style={{ width: "100%" }} />
            </View>
            <View style={styles.weekNameContainer}>
              <Text style={[styles.weekText, { color: theme().color.red }]}>
                日
              </Text>
              <Text style={styles.weekText}>月</Text>
              <Text style={styles.weekText}>火</Text>
              <Text style={styles.weekText}>水</Text>
              <Text style={styles.weekText}>木</Text>
              <Text style={styles.weekText}>金</Text>
              <Text style={[styles.weekText, { color: theme().color.sky }]}>
                土
              </Text>
            </View>
            {this.props.loading ? (
              <ActivityIndicator size="large" color={theme().mode.text} />
            ) : (
              <Calendar
                ref={ref => (this.calendar = ref)}
                style={{
                  backgroundColor: backgroundColors[currentMonth]
                }}
                monthFormat={""}
                minDate={"2019-10-01"}
                maxDate={"2019-12-31"}
                hideDayNames
                hideArrows
                theme={{
                  "stylesheet.calendar.main": {
                    week: {
                      marginTop: 0,
                      flexDirection: "row",
                      justifyContent: "space-between"
                    }
                  },
                  textMonthFontSize: 18,
                  monthTextColor: "#000",
                  textMonthFontWeight: "600",
                  "stylesheet.calendar.header": {
                    header: {
                      height: 0
                    },
                    dayHeader: {
                      fontWeight: "600",
                      paddingBottom: 10,
                      color: "#000"
                    }
                  }
                }}
                dayComponent={({ date }) => {
                  const schedule = this.props.calendars.find(
                    item => item.date === date.dateString
                  );

                  if (schedule) {
                    return (
                      <TouchableOpacity
                        onPress={() =>
                          this.props.onSchedule(schedule.itemId, schedule.title)
                        }
                      >
                        <ImageDay kind={schedule.kind} day={String(date.day)} />
                      </TouchableOpacity>
                    );
                  }

                  return (
                    <TouchableOpacity
                      onPress={() => this.props.onCreate(date.dateString)}
                    >
                      <View
                        style={[
                          styles.itemContainer,
                          {
                            backgroundColor: Color(
                              backgroundColors[currentMonth]
                            )
                              .lighten(0.4)
                              .toString()
                          }
                        ]}
                      >
                        <Text style={styles.dayText}>{date.day}</Text>
                      </View>
                    </TouchableOpacity>
                  );
                }}
              />
            )}
            <View style={styles.calendarBottom} />
          </AnimatedSafeAreaView>
        </GestureRecognizer>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    height: "100%",
    width: "100%"
  },
  headerContainer: {
    justifyContent: "space-between",
    alignItems: "center",
    flexDirection: "row"
  },
  yearContainer: {
    paddingVertical: 20,
    justifyContent: "center"
  },
  year: {
    textAlign: "center",
    color: theme().color.darkGray,
    fontSize: 25,
    fontWeight: "600"
  },
  imageContainer: {
    paddingTop: 15,
    paddingBottom: 15,
    alignItems: "center",
    justifyContent: "center"
  },
  itemContainer: {
    width: width / 7,
    height: width / 7,
    borderTopWidth: 0.5,
    borderRightWidth: 0.5,
    borderLeftWidth: 0.5,
    borderColor: "gray",
    alignItems: "center",
    justifyContent: "center"
  },
  weekNameContainer: {
    marginTop: 7,
    flexDirection: "row",
    justifyContent: "space-around",
    marginHorizontal: 4
  },
  weekText: {
    paddingVertical: 12,
    textAlign: "center",
    alignItems: "center",
    fontSize: 18,
    fontWeight: "600"
  },
  calendarBottom: {
    borderColor: "gray",
    marginHorizontal: 4,
    borderBottomWidth: 1
  },
  dayText: {
    textAlign: "center",
    color: theme().color.gray,
    fontSize: 18,
    fontWeight: "600"
  }
});
