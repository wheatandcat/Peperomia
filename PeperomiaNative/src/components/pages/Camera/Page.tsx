import React, { Component } from "react";
import { Text, View, TouchableOpacity } from "react-native";
import Svg, { Circle } from "react-native-svg";
import * as Permissions from "expo-permissions";
import { Camera } from "expo-camera";
import theme from "../../../config/theme";

export interface Props {
  onPicture: (image: string) => void;
}

export interface State {
  hasCameraPermission: boolean | null;
}

export default class extends Component<Props, State> {
  state = {
    hasCameraPermission: null // カメラ機能の許可
  };
  camera: any;

  // 初期起動時、カメラの使用の権限を取得する。
  async componentDidMount() {
    const { status } = await Permissions.askAsync(Permissions.CAMERA);
    this.setState({
      hasCameraPermission: status === "granted"
    });
  }

  async takePicture() {
    if (!this.camera) {
      return;
    }
    const pictureData = await this.camera.takePictureAsync();

    this.props.onPicture(pictureData.uri);
  }

  render() {
    const { hasCameraPermission } = this.state;

    if (hasCameraPermission === null || hasCameraPermission === false) {
      return (
        <View>
          <Text>カメラの使用が許可されていません。</Text>
        </View>
      );
    }

    return (
      <View
        style={{
          height: "100%"
        }}
      >
        <View
          style={{
            height: "80%"
          }}
        >
          <Camera
            style={{
              flex: 1,
              height: "100%"
            }}
            type={Camera.Constants.Type.back}
            ref={(ref: any) => {
              this.camera = ref;
            }}
          />
        </View>
        <View
          style={{
            alignItems: "center",
            padding: 30
          }}
        >
          <TouchableOpacity onPress={this.takePicture.bind(this)}>
            <Svg height={60} width={60}>
              <Circle
                cx={30}
                cy={30}
                r={25}
                strokeWidth={6}
                stroke={theme().color.gray}
                fill={theme().color.white}
              />
            </Svg>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}
