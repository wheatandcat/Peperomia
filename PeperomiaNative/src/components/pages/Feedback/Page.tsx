import React, { Component } from "react";
import {
  View,
  Text,
  TextInput,
  KeyboardAvoidingView,
  Image,
  Platform,
  InputAccessoryView,
  Keyboard
} from "react-native";
import { Button, Overlay } from "react-native-elements";
import theme from "../../../config/theme";

interface Props {
  isOpen: boolean;
  loading: boolean;
  onFeedback: (message: string) => void;
  onClose: () => void;
}

interface State {
  message: string;
}

export default class extends Component<Props, State> {
  state = {
    message: ""
  };

  render() {
    const inputAccessoryViewID = "feedbackID";

    return (
      <>
        <Overlay isVisible={this.props.isOpen} height={280}>
          <View style={{ alignItems: "center", justifyContent: "center" }}>
            <View style={{ padding: 15 }}>
              <Image
                source={require("../../../img/check.png")}
                style={{ height: 125, width: 125 }}
              />
            </View>
            <Text
              style={{
                fontSize: 15,
                fontWeight: "600",
                paddingVertical: 10,
                paddingHorizontal: 5
              }}
            >
              フィードバックありがとうございます！
            </Text>
            <View style={{ padding: 15 }}>
              <Button
                titleStyle={{ fontWeight: "500" }}
                title="閉じる"
                buttonStyle={{
                  borderRadius: 25,
                  paddingVertical: 8,
                  backgroundColor: theme.color.blue
                }}
                containerStyle={{
                  width: 120
                }}
                onPress={this.props.onClose}
              />
            </View>
          </View>
        </Overlay>
        <View
          style={{ height: "100%", backgroundColor: theme.color.highLightGray }}
        >
          <KeyboardAvoidingView behavior="padding">
            <View>
              <View style={{ backgroundColor: theme.color.highLightGray }}>
                <Text
                  style={{
                    paddingHorizontal: 20,
                    paddingVertical: 10,
                    fontSize: 16,
                    lineHeight: 20
                  }}
                >
                  気になる点や不具合等があれば、こちらからフィードバックお願いします。
                </Text>
              </View>
              <View
                style={{
                  paddingTop: 5,
                  paddingHorizontal: 20,
                  height: 250,
                  backgroundColor: theme.color.white
                }}
              >
                <TextInput
                  placeholder="フィードバックを書く"
                  multiline
                  style={{
                    fontSize: 16,
                    lineHeight: 24,
                    fontWeight: "400"
                  }}
                  onChangeText={message => {
                    this.setState({
                      message
                    });
                  }}
                  testID="inputTextFeedbackMemo"
                  inputAccessoryViewID={inputAccessoryViewID}
                  autoFocus
                >
                  <Text
                    style={{
                      fontSize: 16,
                      lineHeight: 24,
                      fontWeight: "400"
                    }}
                  >
                    {this.state.message}
                  </Text>
                </TextInput>
              </View>
              <Button
                title="フィードバックを送信する"
                titleStyle={{ fontWeight: "600" }}
                buttonStyle={{
                  borderRadius: 0,
                  paddingVertical: 15
                }}
                disabled={this.state.message === "" || this.props.loading}
                onPress={() => this.props.onFeedback(this.state.message)}
                loading={this.props.loading}
              />
            </View>
          </KeyboardAvoidingView>

          {Platform.OS === "ios" && (
            <InputAccessoryView nativeID={inputAccessoryViewID}>
              <View style={{ alignItems: "flex-end" }}>
                <Button
                  buttonStyle={{ width: 80, right: 0, borderRadius: 0 }}
                  onPress={() => Keyboard.dismiss()}
                  title="閉じる"
                />
              </View>
            </InputAccessoryView>
          )}
        </View>
      </>
    );
  }
}
