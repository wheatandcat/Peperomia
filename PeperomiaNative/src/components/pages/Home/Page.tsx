import React from "react";
import { View, Text } from "react-native";
import Cards, { Props as CardsProps } from "../../organisms/Home/Cards";

export interface Props extends CardsProps {
  onCreate: () => void;
}

interface ItemProps extends Props {
  onDelete: (id: string) => void;
}

export default (props: ItemProps) => (
  <View
    style={{
      backgroundColor: "#fff"
    }}
  >
    <View style={{ height: "100%", paddingTop: 3 }}>
      {props.data.length > 0 ? (
        <Cards
          data={props.data}
          loading={false}
          onSchedule={props.onSchedule}
          onDelete={props.onDelete}
        />
      ) : (
        <View
          style={{
            flex: 1,
            justifyContent: "center",
            alignItems: "center"
          }}
        >
          {!props.loading && <Text>プランの登録はありません</Text>}
        </View>
      )}
    </View>
  </View>
);
