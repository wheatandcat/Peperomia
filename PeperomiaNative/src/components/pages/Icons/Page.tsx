import React, { Component } from "react";
import { View, ScrollView, Platform } from "react-native";
import { Input, ListItem, Divider } from "react-native-elements";
import { MaterialIcons } from "@expo/vector-icons";
import {
  ActionSheetProps,
  connectActionSheet
} from "@expo/react-native-action-sheet";
import { KINDS } from "../../../lib/getKind";
import { whenIPhoneSE } from "../../../lib/responsive";
import { IconImage } from "../../atoms";

export interface Props {
  kind: string;
  defaultIcon: boolean;
  photo: boolean;
  onSelectIcon: (kind: string) => void;
  onPhoto: () => void;
  onCamera: () => void;
}

export interface State {
  search: string;
}

class Page extends Component<Props & ActionSheetProps, State> {
  state = {
    search: ""
  };

  onOpenActionSheet = () => {
    this.props.showActionSheetWithOptions(
      {
        options: ["写真を撮影する", "フォトライブラリー", "キャンセル"],
        cancelButtonIndex: 2
      },
      buttonIndex => {
        if (buttonIndex === 0) {
          this.props.onCamera();
        }
        if (buttonIndex === 1) {
          this.props.onPhoto();
        }
      }
    );
  };

  render() {
    const items = Object.entries(KINDS)
      .map(([key, value]) => ({
        kind: key,
        ...value
      }))
      .filter((item: any) => {
        if (this.state.search === "") {
          return true;
        }

        return (
          item.name.includes(this.state.search) ||
          item.kind.includes(this.state.search)
        );
      });

    return (
      <View style={{ backgroundColor: "#ffffff" }}>
        <View
          style={{
            paddingHorizontal: 10,
            height: "7%",
            backgroundColor: "#eeeeee",
            alignItems: "center"
          }}
        >
          <Input
            placeholder="検索"
            leftIcon={{ type: "MaterialIcons", name: "search", color: "#888" }}
            inputContainerStyle={{
              backgroundColor: "#cccccc",
              borderBottomWidth: 0,
              borderRadius: 10,
              height: Platform.OS === "ios" ? whenIPhoneSE(30, 45) : 30
            }}
            leftIconContainerStyle={{
              marginRight: 20
            }}
            containerStyle={{
              paddingTop: Platform.OS === "ios" ? 0 : 5
            }}
            onChangeText={text => this.setState({ search: text })}
          />
        </View>
        <Divider />
        <ScrollView style={{ width: "100%", height: "80%", paddingLeft: 15 }}>
          {items.map((item: any, i: number) => (
            <ListItem
              key={i}
              title={
                !this.props.defaultIcon && item.name === "地球"
                  ? "アイコンなし"
                  : item.name
              }
              onPress={() => this.props.onSelectIcon(item.kind)}
              leftIcon={
                <IconImage
                  {...item}
                  size={20}
                  opacity={1.0}
                  defaultIcon={this.props.defaultIcon}
                />
              }
              rightIcon={
                <View>
                  {this.props.kind === item.kind ? (
                    <MaterialIcons
                      name="check"
                      color="#32cd32"
                      size={20}
                      style={{ paddingRight: 5 }}
                    />
                  ) : null}
                </View>
              }
              bottomDivider
            />
          ))}
        </ScrollView>
        <View
          style={{
            backgroundColor: "#eee",
            height: "8%",
            alignItems: "flex-end"
          }}
        />
      </View>
    );
  }
}

export default connectActionSheet(Page);
