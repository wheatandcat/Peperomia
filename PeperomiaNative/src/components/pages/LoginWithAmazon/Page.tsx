import React from "react";
import { View, Text, Image, Linking } from "react-native";
import { Button, Divider } from "react-native-elements";
import EStyleSheet from "react-native-extended-stylesheet";

type Props = {
  loading: boolean;
  linked: boolean;
  onAmazonLogin: () => void;
};

export default (props: Props) => {
  return (
    <View style={styles.root}>
      {props.linked ? (
        <View style={styles.textContainer}>
          <Divider />
          <Text style={styles.text}>連携済みです</Text>
          <Divider />
          <View style={styles.alexaSkillContainer}>
            <View style={styles.log}>
              <Image
                source={require("../../../img/alexa_skill.png")}
                style={{ height: 80, width: 80 }}
              />
            </View>
            <View style={styles.log}>
              <Button
                title="Alexa Skillを確認する"
                type="clear"
                onPress={() => Linking.openURL("https://peperomia.app/")}
              />
            </View>
          </View>
          <Divider />
        </View>
      ) : (
        <Button
          title="Alexaスキルとアカウントをリンクする"
          onPress={props.onAmazonLogin}
          loading={props.loading}
          disabled={props.loading}
        />
      )}
    </View>
  );
};

const styles = EStyleSheet.create({
  root: {
    backgroundColor: "$background",
    height: "100%",
    width: "100%",
    alignItems: "center",
    justifyContent: "center"
  },
  textContainer: {
    width: "100%"
  },
  text: {
    color: "$text",
    fontSize: 18,
    fontWeight: "600",
    textAlign: "center",
    paddingVertical: 8
  },
  alexaSkillContainer: {
    flexDirection: "row",
    alignItems: "center"
  },
  log: {
    padding: 15
  }
});
