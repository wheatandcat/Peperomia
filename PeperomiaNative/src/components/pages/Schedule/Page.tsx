import React, { Component } from 'react';
import { View, Alert } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import Cards, { Props as CardsProps } from '../../organisms/Schedule/Cards';
import ActionButton from '../../molecules/Schedule/ActionButton';

export interface Props extends CardsProps {
  onAdd: () => void;
  onSort: () => void;
  onDelete: () => void;
}

export default class extends Component<Props> {
  onDelete = () => {
    Alert.alert(
      '本当に削除しますか？',
      '',
      [
        {
          text: 'キャンセル',
          style: 'cancel',
        },
        {
          text: '削除する',
          onPress: () => {
            this.props.onDelete();
          },
        },
      ],
      { cancelable: false }
    );
  };

  render() {
    return (
      <View style={styles.root}>
        <Cards
          data={this.props.data}
          onScheduleDetail={this.props.onScheduleDetail}
        />
        <View style={styles.footer}>
          <ActionButton
            onAdd={this.props.onAdd}
            onSort={this.props.onSort}
            onDelete={this.onDelete}
          />
        </View>
      </View>
    );
  }
}

const styles = EStyleSheet.create({
  root: {
    backgroundColor: '$background',
    height: '100%',
  },
  footer: {
    width: 60,
    height: 60,
    position: 'absolute',
    bottom: 50,
    right: 40,
  },
});
