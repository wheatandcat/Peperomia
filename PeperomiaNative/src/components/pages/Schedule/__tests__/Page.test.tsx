import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Page from '../Page';
import { mockData } from './mockData';

describe('components/pages/Schedule/Page.tsx', () => {
  let wrapper: ShallowWrapper;

  const propsData = () => ({
    data: mockData,
    loading: false,
    onAdd: jest.fn(),
    onSort: jest.fn(),
    onDelete: jest.fn(),
    onScheduleDetail: jest.fn(),
  });

  beforeEach(() => {
    wrapper = shallow(<Page {...propsData()} />);
  });

  it('正常に表示されている', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
