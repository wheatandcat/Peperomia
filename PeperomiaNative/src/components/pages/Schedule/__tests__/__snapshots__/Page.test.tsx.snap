// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Schedule/Page.tsx 正常に表示されている 1`] = `
<View>
  <_default
    data={
      Array [
        Object {
          "id": "1",
          "itemId": "1",
          "kind": "park",
          "memo": "",
          "moveMinutes": 30,
          "place": "",
          "priority": 1,
          "title": "新宿駅",
          "url": "",
        },
      ]
    }
    loading={false}
    onScheduleDetail={[MockFunction]}
  />
  <View>
    <_default
      onAdd={[MockFunction]}
      onDelete={[Function]}
      onSort={[MockFunction]}
    />
  </View>
</View>
`;
